### PR TITLE
Sprint 2 - PR develop to master

### PR DESCRIPTION
**Pull request by MarineMontailler**

### CODE REVIEW

File: src/components/LoginPageLayout.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+const LogPage = styled.div`
+
+background-image: url(${img});
+background-size:cover;
+background-position:top ;
+width:375px;
+height:812px;
+color:white;
+
+input,button {
```
Comment (bugsfunny) [inline src/components/LoginPageLayout.js]: Please make Button and input component

File: src/components/MainPage/CocktailRate.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import styled from "styled-components";
+
+
+const CocktailRate = styled.div`
+p{
```
Comment (bugsfunny) [inline src/components/MainPage/CocktailRate.js]: very strange practice

File: src/components/MainPage/Header.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+const Header = styled.header`
+  grid-area: head;
+  display:flex;
+  text-align:center;
+  align-items:center;
+  margin-left: 0px;
+  
+  
+  h1 {
```
Comment (bugsfunny) [inline src/components/MainPage/Header.js]: make title component ?

File: src/components/MainPage/Main.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+  
+  
+  display:flex;
+  flex-direction:column;
+  overflow-y: scroll;
+  margin-bottom:100px;
+  align-items:center;
+  margin-top:50px;
+
+  .cocktailList {
```
Comment (bugsfunny) [inline src/components/MainPage/Main.js]: Make a component

File: src/components/MainPage/Main.js [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+flex-direction:column;
+align-items:center;
+margin-top:30px;
+
+
+
+
+  };
+
+  .cocktailInfos {
```
Comment (bugsfunny) [inline src/components/MainPage/Main.js]: Make a component

File: src/components/MainPage/Main.js [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+    display:flex  ;
+flex-direction:row;
+justify-content:space-around;
+align-items:center;
+
+};
+
+.cocktailTitle {
```
Comment (bugsfunny) [inline src/components/MainPage/Main.js]: Make a component

File: src/components/MainPage/Main.js [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+flex:3;
+margin-right:70%;
+};
+
+.cocktailRate {
```
Comment (bugsfunny) [inline src/components/MainPage/Main.js]: Make a component

File: src/components/MainPage/Main.js [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+flex:1;
+display:flex;
+margin-right:30%;
+
+
+
+};
+
+.clickable {
```
Comment (bugsfunny) [inline src/components/MainPage/Main.js]: ??

File: src/components/MainPage/Search.js [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+
+/* Base styling */
+
+
+hr {
+  width:100%;
+  margin-top:20px;
+}
+
+button {
```
Comment (bugsfunny) [inline src/components/MainPage/Search.js]: Make a component

File: src/components/MainPage/Search.js [hunk 10]
```diff
@@ -0,0 +1,105 @@
...
+}
+
+.button-secondary {
+            color: white;
+            border-radius: 4px;
+            text-shadow: 0 1px 1px rgba(0, 0, 0, 0.2);
+            background: #009ce7;
+        }
+
+.searchBar {
```
Comment (bugsfunny) [inline src/components/MainPage/Search.js]: Make a component

File: src/components/MainPage/Search.js [hunk 11]
```diff
@@ -0,0 +1,105 @@
...
+display:flex;
+justify-content:center;
+align-items:center;
+margin-bottom:15px;
+
+}
+
+.searchTags {
```
Comment (bugsfunny) [inline src/components/MainPage/Search.js]: Make a component

File: src/components/MainPage/Search.js [hunk 12]
```diff
@@ -0,0 +1,105 @@
...
+  display:flex;
+  justify-content:flex-start;
+  align-self:center;
+  margin-left:-30px;
+}
+
+.search {
```
Comment (bugsfunny) [inline src/components/MainPage/Search.js]: Make a component

File: src/containers/DisplaySearch.js [hunk 13]
```diff
@@ -0,0 +1,30 @@
+import React, {useState} from "react";
+import 'font-awesome/css/font-awesome.min.css';
+import TextField from "@material-ui/core/TextField";
+import Autocomplete from "@material-ui/lab/Autocomplete";
+import liste_ingredient from './liste_ingredient.js';
+
+function DisplaySearch(){
+const [value, setValue] = useState([]);
+const Search = ({ value }) =>
```
Comment (bugsfunny) [inline src/containers/DisplaySearch.js]: do you follow a doc... it seems a little strange

File: src/containers/LoginPage/LoginPage.js [hunk 14]
```diff
@@ -0,0 +1,64 @@
...
+const LoginPage = props => {
+
+
+ 
+  
+
+
+
+  
+
```
Comment (bugsfunny) [inline src/containers/LoginPage/LoginPage.js]: Please use prettier before push

File: src/containers/Navbar.js [hunk 15]
```diff
@@ -0,0 +1,22 @@
+import React from "react";
+import { Link } from "react-router-dom";
+
+const Navbar = () => {
+  return (
+    <>
+      <i className="fa fa-2x fa-refresh" aria-hidden="true"></i>
```
Comment (bugsfunny) [inline src/containers/Navbar.js]: you forget it ?

File: src/containers/PrivateRoute.js [hunk 16]
```diff
@@ -0,0 +1,27 @@
...
+
+function PrivateRoute({ component: Component, ...rest }) {
+
+    const fakeAuth=useContext(AuthentContext);
+    console.log("Authentifié avant redirection Main?"+fakeAuth.isAuthenticated)
+    return (
+      <Route
+        {...rest}
+        render={(props) =>
+          true ? (
```
Comment (bugsfunny) [inline src/containers/PrivateRoute.js]: ah ah this is the hack !!!

File: src/containers/RecipePage.js [hunk 17]
```diff
@@ -0,0 +1,71 @@
...
+  const getIngredients = (dataRecipe) => {
+    let ingredients = [];
+    for (let i = 1; i < 16; i++) {
+      ingredients.push({
+        name: dataRecipe["strIngredient" + i],
+        measure: dataRecipe["strMeasure" + i],
+        id: i,
+      });
+    }
+    return ingredients.filter((ingredient) => ingredient.name != null);
```
Comment (bugsfunny) [inline src/containers/RecipePage.js]: i think you can put a condition in the for loop

File: src/containers/RecipePage.js [hunk 18]
```diff
@@ -0,0 +1,71 @@
...
+  };
+
+  let listOfIngredients = [];
+  if (dataRecipe !== undefined && dataRecipe !== null) {
```
Comment (bugsfunny) [inline src/containers/RecipePage.js]: i prefer you test if null (see comment after) , you do not have to test undefined if you set default state value at null

File: src/containers/RecipePage.js [hunk 19]
```diff
@@ -0,0 +1,71 @@
...
+        <Ingredients
+          item={item}
+          key={item.id}
+          addIngredient={props.addIngredient}
+        />
+      );
+    });
+  }
+
+  if (loading) {
```
Comment (bugsfunny) [inline src/containers/RecipePage.js:61]: if data recipe is null ?

File: src/containers/RecipePage.js [hunk 20]
```diff
@@ -0,0 +1,71 @@
...
+    return <h2>Loading...</h2>;
+  }
+
+  return (
```
Comment (bugsfunny) [inline src/containers/RecipePage.js:65]: I think you can make a component

File: src/containers/ShoppingListPage.js [hunk 21]
```diff
@@ -0,0 +1,34 @@
...
+import Navbar from "./Navbar";
+import ShoppingListContent from "./ShoppingListContent";
+
+const ShoppingListPage = (props) => {
+  return (
+    <div>
+      <GridLayout>
+        <Header>
+          <Link to="/main">
+            <img
```
Comment (bugsfunny) [inline src/containers/ShoppingListPage.js]: Make a component

File: src/containers/liste_ingredient.js [hunk 22]
```diff
@@ -0,0 +1,203 @@
+const listeIngredient = [{
```
Comment (bugsfunny) [inline src/containers/liste_ingredient.js:1]: List of ingredient should be in the API not in a file ?

File: src/components/MainPage/Header.js (hunk 3 above)
Comment (grldf) [reply src/components/MainPage/Header.js]: change Ok

File: src/components/MainPage/Main.js (hunk 7 above)
Comment (grldf) [reply src/components/MainPage/Main.js]: change ok

File: src/components/MainPage/Main.js (hunk 8 above)
Comment (grldf) [reply src/components/MainPage/Main.js]: test first week, change ok

File: src/containers/ShoppingListPage.js (hunk 21 above)
Comment (grldf) [reply src/containers/ShoppingListPage.js]: change ok

File: src/components/MainPage/Main.js (hunk 4 above)
Comment (grldf) [reply src/components/MainPage/Main.js]: change ok

File: src/components/MainPage/Main.js (hunk 5 above)
Comment (grldf) [reply src/components/MainPage/Main.js]: change ok

File: src/components/MainPage/Main.js (hunk 6 above)
Comment (grldf) [reply src/components/MainPage/Main.js]: change ok

File: src/containers/LoginPage/LoginPage.js (hunk 14 above)
Comment (MarineMontailler) [reply src/containers/LoginPage/LoginPage.js]: ok prettier is now install

**Pull request closed without merging**